lib/datasource/remote/note_remote_datasource.dart

```diff
@@ -0,0 +1,73 @@
+import 'dart:developer';
+
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:notes_app/models/note_model.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+
+final noteRemoteProvider = Provider<NoteRemoteDatasource>(
+  (ref) => NoteRemoteDatasource()
+);
+
+class NoteRemoteDatasource {
+  NoteRemoteDatasource._();
+  static final NoteRemoteDatasource _instance = NoteRemoteDatasource._();
```

Comment:
Same with https://github.com/HuyVoWakumo/NotesApp/pull/3/files#r1687354645
Please check other places to fix